Tests/ApolloCodegenTests/CodeGenAST/ASTSelectionSetTests.swift

```diff
@@ -0,0 +1,2415 @@
+import XCTest
+import Nimble
+import OrderedCollections
+@testable import ApolloCodegenLib
+import ApolloCodegenTestSupport
+
+class ASTSelectionSetTests: XCTestCase {
+
+  var mockCompilationResult: CompilationResult!
+
+  override func setUp() {
+    super.setUp()
+    mockCompilationResult = CompilationResult.mock()
+  }
+
+  override func tearDown() {
+    mockCompilationResult = nil
+    super.tearDown()
+  }
+
+  // MARK: - Children Computation
+
+  // MARK: Children - Fragment Type
+
+  /// Example:
+  /// query {
+  ///  allAnimals {
+  ///    ...AnimalDetails
+  /// }
+  ///
+  /// fragment AnimalDetails on Animal {
+  ///   species
+  /// }
+  /// Expected:
+  /// Birds.children should not include a type case for asAnimal
```

Comment:
this is *extremely* helpful in figuring out what the test is doing, however I'm not seeing where the the subject type is being set as `Birds` - where is that?

Reply:
Whoops, this should be `Animal` and it's set on line 43. My mistake!